opencog/atomspace/AtomTable.h

```diff
@@ -219,10 +219,20 @@ class AtomTable
                        bool subclass = false) const
     {
         std::lock_guard<std::recursive_mutex> lck(_mtx);
-        return std::copy_if(typeIndex.begin(type, subclass),
-                            typeIndex.end(),
-                            result,
-                            isDefined);
+
+        // We're going to use this one handle so we don't construct/destruct
+        // googles of temporaries in the process of doing the copies.
+        Handle temp_handle;
+        std::for_each(typeIndex.begin(type, subclass),
+                      typeIndex.end(),
+            [&](UUID uuid)->void {
+                if (UNDEFINED_UUID == uuid) return;
```

Comment:
I was wondering about that. Even added a ??? to one of the files hoping you'd answer it.